qt/scientific_interfaces/Inelastic/Analysis/IndirectFitDataView.cpp

```diff
@@ -49,7 +53,26 @@ class ExcludeRegionDelegate : public QItemDelegate {
   }
 };
 
-QString makeNumber(double d) { return QString::number(d, 'g', 6); }
+class NumericInputDelegate : public QStyledItemDelegate {
+
+public:
+  QWidget *createEditor(QWidget *parent, const QStyleOptionViewItem &, const QModelIndex &) const override {
+
+    auto lineEdit = std::make_unique<QLineEdit>(parent);
+    auto validator = std::make_unique<QDoubleValidator>(parent);
+
+    validator->setDecimals(6);
+    validator->setNotation(QDoubleValidator::StandardNotation);
+    lineEdit->setValidator(validator.release());
+
+    return lineEdit.release();
```

Comment:
Nice thought to use unique_ptrs - you are right to think it is almost always better to use these instead of raw pointers. The main reason being that with raw pointers you need to remember to free the memory when you go out-of-scope, and with unique_ptrs you don't

Having said this, QObjects are a bit of an exception to this rule, and it is preferable to use raw pointers for them. The reason for this is because QObjects are owned by the QApplication, and so we do not want the lifetime of the QObjects to be managed by anything other than the QApplication itself, or a parent widget. In this case, it is preferable to do something like this:
```suggestion
    auto lineEdit = new QLineEdit(parent);
    auto validator = new QDoubleValidator(parent);

    validator->setDecimals(6);
    validator->setNotation(QDoubleValidator::StandardNotation);
    lineEdit->setValidator(validator);

    return lineEdit;
```

Reply:
Yes that is right, `release` will remove the ownership responsibility of the unique pointer (basically the same as defining a raw pointer in the first place). The lifetime of the widgets is then most likely managed by their parent widget